react/components/EditorContainer/Topbar/ContextSelectors/BindingSelector.tsx

```diff
@@ -0,0 +1,57 @@
+import React from 'react'
+import { Dropdown, Tooltip } from 'vtex.styleguide'
+import { useIntl } from 'react-intl'
+
+import { useEditorContext } from '../../../EditorContext'
+import { DropdownOptions, DropdownValue } from './typings'
+
+interface Props {
+  isDisabled: boolean
+  onChange: React.ChangeEventHandler<HTMLSelectElement>
+  options: DropdownOptions
+  value: DropdownValue
+}
+
+const BindingSelector: React.FC<Props> = ({
+  isDisabled,
+  onChange,
+  options,
+  value,
+}) => {
+  const Selector = () => (
```

Comment:
I think it's a bad practice to declare a component inside another component because it may cause unintended performance problems, since you are defining a new component every time the function is called (much like arrow functions in callbacks).

I suggest moving this component outside `BindingSelector` scope, what do you think?

Reply:
How about using memoization then?

Reply:
Memoization solves the issue as well